can't compile grpc gem on MacOS BigSur

### What version of gRPC and what language are you using?

gprc 1.30.2 with ruby 2.7.2

### What operating system (Linux, Windows,...) and version?

Mac OS BigSur 11.0.1 (20B50)

### What runtime / compiler are you using (e.g. python version or version of gcc)

* ruby 2.7.2p137 (2020-10-01 revision 5445e04352) [x86_64-darwin19]
* Bundler version 2.1.4
* gcc:
```
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Apple clang version 12.0.0 (clang-1200.0.32.27)
Target: x86_64-apple-darwin20.1.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```


### What did you do?

bundle install with grpc locked to 1.30.2

### What did you expect to see?

it should compile/install without issues

### What did you see instead?

Here is the full log from the compilation:

```
When trying to install current GitLab using grpc-1.30.2, I get this compilation output:
current directory: /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc
/Users/brodock/.rvm/rubies/ruby-2.7.2/bin/ruby -I /Users/brodock/.rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0 -r ./siteconf20201202-93469-1d8riw9.rb extconf.rb --with-cflags\=-Wno-error\=implicit-function-declaration
Building internal gRPC into /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/libs/opt
[C]       Compiling third_party/zlib/adler32.c
[C]       Compiling third_party/zlib/compress.c
mkdir -p `dirname /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/adler32.o`
[C]       Compiling third_party/zlib/crc32.c
mkdir -p `dirname /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/crc32.o`
[C]       Compiling third_party/zlib/deflate.c
mkdir -p `dirname /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/deflate.o`
mkdir -p `dirname /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/compress.o`
[C]       Compiling third_party/zlib/gzclose.c
mkdir -p `dirname /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/gzclose.o`
[C]       Compiling third_party/zlib/gzlib.c
mkdir -p `dirname /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/gzlib.o`
[C]       Compiling third_party/zlib/gzread.c
[C]       Compiling third_party/zlib/gzwrite.c
mkdir -p `dirname /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/gzwrite.o`
[C]       Compiling third_party/zlib/infback.c
mkdir -p `dirname /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/gzread.o`
mkdir -p `dirname /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/infback.o`
[C]       Compiling third_party/zlib/inffast.c
mkdir -p `dirname /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/inffast.o`
[C]       Compiling third_party/zlib/inflate.c
[C]       Compiling third_party/zlib/inftrees.c
[C]       Compiling third_party/zlib/trees.c
[C]       Compiling third_party/zlib/uncompr.c
[C]       Compiling third_party/zlib/zutil.c
mkdir -p `dirname /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/zutil.o`
gcc -Ithird_party/googletest/googletest/include -Ithird_party/googletest/googlemock/include -Ithird_party/boringssl-with-bazel/src/include -Ithird_party/cares -Ithird_party/cares/cares -Ithird_party/address_sorting/include -DGPR_BACKWARDS_COMPATIBILITY_MODE -g -Wall -Wextra -DOSATOMIC_USE_INLINED=1 -Ithird_party/abseil-cpp -Ithird_party/upb -Isrc/core/ext/upb-generated -O2 -Wframe-larger-than=16384 -fPIC -I. -Iinclude -I/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/gens -I/usr/local/include -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -arch i386 -arch x86_64  -Ithird_party/zlib -std=c99  -g   -fvisibility=hidden -MMD -MF /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/inffast.dep -c -o /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/inffast.o third_party/zlib/inffast.c
mkdir -p `dirname /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/uncompr.o`
mkdir -p `dirname /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/trees.o`
mkdir -p `dirname /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/inftrees.o`
gcc -Ithird_party/googletest/googletest/include -Ithird_party/googletest/googlemock/include -Ithird_party/boringssl-with-bazel/src/include -Ithird_party/cares -Ithird_party/cares/cares -Ithird_party/address_sorting/include -DGPR_BACKWARDS_COMPATIBILITY_MODE -g -Wall -Wextra -DOSATOMIC_USE_INLINED=1 -Ithird_party/abseil-cpp -Ithird_party/upb -Isrc/core/ext/upb-generated -O2 -Wframe-larger-than=16384 -fPIC -I. -Iinclude -I/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/gens -I/usr/local/include -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -arch i386 -arch x86_64  -Ithird_party/zlib -std=c99  -g   -fvisibility=hidden -MMD -MF /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/infback.dep -c -o /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/infback.o third_party/zlib/infback.c
gcc -Ithird_party/googletest/googletest/include -Ithird_party/googletest/googlemock/include -Ithird_party/boringssl-with-bazel/src/include -Ithird_party/cares -Ithird_party/cares/cares -Ithird_party/address_sorting/include -DGPR_BACKWARDS_COMPATIBILITY_MODE -g -Wall -Wextra -DOSATOMIC_USE_INLINED=1 -Ithird_party/abseil-cpp -Ithird_party/upb -Isrc/core/ext/upb-generated -O2 -Wframe-larger-than=16384 -fPIC -I. -Iinclude -I/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/gens -I/usr/local/include -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -arch i386 -arch x86_64  -Ithird_party/zlib -std=c99  -g   -fvisibility=hidden -MMD -MF /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/gzread.dep -c -o /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/gzread.o third_party/zlib/gzread.c
gcc -Ithird_party/googletest/googletest/include -Ithird_party/googletest/googlemock/include -Ithird_party/boringssl-with-bazel/src/include -Ithird_party/cares -Ithird_party/cares/cares -Ithird_party/address_sorting/include -DGPR_BACKWARDS_COMPATIBILITY_MODE -g -Wall -Wextra -DOSATOMIC_USE_INLINED=1 -Ithird_party/abseil-cpp -Ithird_party/upb -Isrc/core/ext/upb-generated -O2 -Wframe-larger-than=16384 -fPIC -I. -Iinclude -I/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/gens -I/usr/local/include -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -arch i386 -arch x86_64  -Ithird_party/zlib -std=c99  -g   -fvisibility=hidden -MMD -MF /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/gzwrite.dep -c -o /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/gzwrite.o third_party/zlib/gzwrite.c
mkdir -p `dirname /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/inflate.o`
gcc -Ithird_party/googletest/googletest/include -Ithird_party/googletest/googlemock/include -Ithird_party/boringssl-with-bazel/src/include -Ithird_party/cares -Ithird_party/cares/cares -Ithird_party/address_sorting/include -DGPR_BACKWARDS_COMPATIBILITY_MODE -g -Wall -Wextra -DOSATOMIC_USE_INLINED=1 -Ithird_party/abseil-cpp -Ithird_party/upb -Isrc/core/ext/upb-generated -O2 -Wframe-larger-than=16384 -fPIC -I. -Iinclude -I/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/gens -I/usr/local/include -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -arch i386 -arch x86_64  -Ithird_party/zlib -std=c99  -g   -fvisibility=hidden -MMD -MF /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/gzlib.dep -c -o /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/gzlib.o third_party/zlib/gzlib.c
gcc -Ithird_party/googletest/googletest/include -Ithird_party/googletest/googlemock/include -Ithird_party/boringssl-with-bazel/src/include -Ithird_party/cares -Ithird_party/cares/cares -Ithird_party/address_sorting/include -DGPR_BACKWARDS_COMPATIBILITY_MODE -g -Wall -Wextra -DOSATOMIC_USE_INLINED=1 -Ithird_party/abseil-cpp -Ithird_party/upb -Isrc/core/ext/upb-generated -O2 -Wframe-larger-than=16384 -fPIC -I. -Iinclude -I/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/gens -I/usr/local/include -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -arch i386 -arch x86_64  -Ithird_party/zlib -std=c99  -g   -fvisibility=hidden -MMD -MF /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/gzclose.dep -c -o /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/gzclose.o third_party/zlib/gzclose.c
gcc -Ithird_party/googletest/googletest/include -Ithird_party/googletest/googlemock/include -Ithird_party/boringssl-with-bazel/src/include -Ithird_party/cares -Ithird_party/cares/cares -Ithird_party/address_sorting/include -DGPR_BACKWARDS_COMPATIBILITY_MODE -g -Wall -Wextra -DOSATOMIC_USE_INLINED=1 -Ithird_party/abseil-cpp -Ithird_party/upb -Isrc/core/ext/upb-generated -O2 -Wframe-larger-than=16384 -fPIC -I. -Iinclude -I/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/gens -I/usr/local/include -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -arch i386 -arch x86_64  -Ithird_party/zlib -std=c99  -g   -fvisibility=hidden -MMD -MF /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/compress.dep -c -o /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/compress.o third_party/zlib/compress.c
gcc -Ithird_party/googletest/googletest/include -Ithird_party/googletest/googlemock/include -Ithird_party/boringssl-with-bazel/src/include -Ithird_party/cares -Ithird_party/cares/cares -Ithird_party/address_sorting/include -DGPR_BACKWARDS_COMPATIBILITY_MODE -g -Wall -Wextra -DOSATOMIC_USE_INLINED=1 -Ithird_party/abseil-cpp -Ithird_party/upb -Isrc/core/ext/upb-generated -O2 -Wframe-larger-than=16384 -fPIC -I. -Iinclude -I/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/gens -I/usr/local/include -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -arch i386 -arch x86_64  -Ithird_party/zlib -std=c99  -g   -fvisibility=hidden -MMD -MF /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/deflate.dep -c -o /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/deflate.o third_party/zlib/deflate.c
gcc -Ithird_party/googletest/googletest/include -Ithird_party/googletest/googlemock/include -Ithird_party/boringssl-with-bazel/src/include -Ithird_party/cares -Ithird_party/cares/cares -Ithird_party/address_sorting/include -DGPR_BACKWARDS_COMPATIBILITY_MODE -g -Wall -Wextra -DOSATOMIC_USE_INLINED=1 -Ithird_party/abseil-cpp -Ithird_party/upb -Isrc/core/ext/upb-generated -O2 -Wframe-larger-than=16384 -fPIC -I. -Iinclude -I/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/gens -I/usr/local/include -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -arch i386 -arch x86_64  -Ithird_party/zlib -std=c99  -g   -fvisibility=hidden -MMD -MF /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/crc32.dep -c -o /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/crc32.o third_party/zlib/crc32.c
gcc -Ithird_party/googletest/googletest/include -Ithird_party/googletest/googlemock/include -Ithird_party/boringssl-with-bazel/src/include -Ithird_party/cares -Ithird_party/cares/cares -Ithird_party/address_sorting/include -DGPR_BACKWARDS_COMPATIBILITY_MODE -g -Wall -Wextra -DOSATOMIC_USE_INLINED=1 -Ithird_party/abseil-cpp -Ithird_party/upb -Isrc/core/ext/upb-generated -O2 -Wframe-larger-than=16384 -fPIC -I. -Iinclude -I/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/gens -I/usr/local/include -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -arch i386 -arch x86_64  -Ithird_party/zlib -std=c99  -g   -fvisibility=hidden -MMD -MF /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/adler32.dep -c -o /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/adler32.o third_party/zlib/adler32.c
gcc -Ithird_party/googletest/googletest/include -Ithird_party/googletest/googlemock/include -Ithird_party/boringssl-with-bazel/src/include -Ithird_party/cares -Ithird_party/cares/cares -Ithird_party/address_sorting/include -DGPR_BACKWARDS_COMPATIBILITY_MODE -g -Wall -Wextra -DOSATOMIC_USE_INLINED=1 -Ithird_party/abseil-cpp -Ithird_party/upb -Isrc/core/ext/upb-generated -O2 -Wframe-larger-than=16384 -fPIC -I. -Iinclude -I/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/gens -I/usr/local/include -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -arch i386 -arch x86_64  -Ithird_party/zlib -std=c99  -g   -fvisibility=hidden -MMD -MF /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/inflate.dep -c -o /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/inflate.o third_party/zlib/inflate.c
gcc -Ithird_party/googletest/googletest/include -Ithird_party/googletest/googlemock/include -Ithird_party/boringssl-with-bazel/src/include -Ithird_party/cares -Ithird_party/cares/cares -Ithird_party/address_sorting/include -DGPR_BACKWARDS_COMPATIBILITY_MODE -g -Wall -Wextra -DOSATOMIC_USE_INLINED=1 -Ithird_party/abseil-cpp -Ithird_party/upb -Isrc/core/ext/upb-generated -O2 -Wframe-larger-than=16384 -fPIC -I. -Iinclude -I/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/gens -I/usr/local/include -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -arch i386 -arch x86_64  -Ithird_party/zlib -std=c99  -g   -fvisibility=hidden -MMD -MF /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/inftrees.dep -c -o /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/inftrees.o third_party/zlib/inftrees.c
gcc -Ithird_party/googletest/googletest/include -Ithird_party/googletest/googlemock/include -Ithird_party/boringssl-with-bazel/src/include -Ithird_party/cares -Ithird_party/cares/cares -Ithird_party/address_sorting/include -DGPR_BACKWARDS_COMPATIBILITY_MODE -g -Wall -Wextra -DOSATOMIC_USE_INLINED=1 -Ithird_party/abseil-cpp -Ithird_party/upb -Isrc/core/ext/upb-generated -O2 -Wframe-larger-than=16384 -fPIC -I. -Iinclude -I/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/gens -I/usr/local/include -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -arch i386 -arch x86_64  -Ithird_party/zlib -std=c99  -g   -fvisibility=hidden -MMD -MF /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/trees.dep -c -o /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/trees.o third_party/zlib/trees.c
gcc -Ithird_party/googletest/googletest/include -Ithird_party/googletest/googlemock/include -Ithird_party/boringssl-with-bazel/src/include -Ithird_party/cares -Ithird_party/cares/cares -Ithird_party/address_sorting/include -DGPR_BACKWARDS_COMPATIBILITY_MODE -g -Wall -Wextra -DOSATOMIC_USE_INLINED=1 -Ithird_party/abseil-cpp -Ithird_party/upb -Isrc/core/ext/upb-generated -O2 -Wframe-larger-than=16384 -fPIC -I. -Iinclude -I/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/gens -I/usr/local/include -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -arch i386 -arch x86_64  -Ithird_party/zlib -std=c99  -g   -fvisibility=hidden -MMD -MF /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/uncompr.dep -c -o /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/uncompr.o third_party/zlib/uncompr.c
gcc -Ithird_party/googletest/googletest/include -Ithird_party/googletest/googlemock/include -Ithird_party/boringssl-with-bazel/src/include -Ithird_party/cares -Ithird_party/cares/cares -Ithird_party/address_sorting/include -DGPR_BACKWARDS_COMPATIBILITY_MODE -g -Wall -Wextra -DOSATOMIC_USE_INLINED=1 -Ithird_party/abseil-cpp -Ithird_party/upb -Isrc/core/ext/upb-generated -O2 -Wframe-larger-than=16384 -fPIC -I. -Iinclude -I/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/gens -I/usr/local/include -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -arch i386 -arch x86_64  -Ithird_party/zlib -std=c99  -g   -fvisibility=hidden -MMD -MF /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/zutil.dep -c -o /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/zutil.o third_party/zlib/zutil.c
[C]       Compiling third_party/cares/cares/ares__close_sockets.c
third_party/zlib/gzwrite.c:89:20: warning: implicit declaration of function 'write' is invalid in C99 [-Wimplicit-function-declaration]
            writ = write(state->fd, strm->next_in, put);
                   ^
third_party/zlib/gzread.c:35:15: warning: implicit declaration of function 'read' is invalid in C99 [-Wimplicit-function-declaration]
        ret = read(state->fd, buf + *have, get);
              ^
third_party/zlib/gzwrite.c:110:24: warning: implicit declaration of function 'write' is invalid in C99 [-Wimplicit-function-declaration]
                writ = write(state->fd, state->x.next, put);
                       ^
third_party/zlib/gzlib.c:252:9: warning: implicit declaration of function 'lseek' is invalid in C99 [-Wimplicit-function-declaration]
        LSEEK(state->fd, 0, SEEK_END);  /* so gzoffset() is correct */
        ^
third_party/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
third_party/zlib/gzlib.c:258:24: warning: implicit declaration of function 'lseek' is invalid in C99 [-Wimplicit-function-declaration]
        state->start = LSEEK(state->fd, 0, SEEK_CUR);
                       ^
third_party/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
third_party/zlib/gzlib.c:359:9: warning: implicit declaration of function 'lseek' is invalid in C99 [-Wimplicit-function-declaration]
    if (LSEEK(state->fd, state->start, SEEK_SET) == -1)
        ^
third_party/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
third_party/zlib/gzlib.c:400:15: warning: implicit declaration of function 'lseek' is invalid in C99 [-Wimplicit-function-declaration]
        ret = LSEEK(state->fd, offset - state->x.have, SEEK_CUR);
              ^
third_party/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
third_party/zlib/gzlib.c:496:14: warning: implicit declaration of function 'lseek' is invalid in C99 [-Wimplicit-function-declaration]
    offset = LSEEK(state->fd, 0, SEEK_CUR);
             ^
third_party/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
third_party/zlib/gzwrite.c:661:9: warning: implicit declaration of function 'close' is invalid in C99 [-Wimplicit-function-declaration]
    if (close(state->fd) == -1)
        ^
third_party/zlib/gzread.c:651:11: warning: implicit declaration of function 'close' is invalid in C99 [-Wimplicit-function-declaration]
    ret = close(state->fd);
          ^
mkdir -p `dirname /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/cares/cares/ares__close_sockets.o`
gcc -Ithird_party/googletest/googletest/include -Ithird_party/googletest/googlemock/include -Ithird_party/boringssl-with-bazel/src/include -Ithird_party/cares -Ithird_party/cares/cares -Ithird_party/address_sorting/include -DGPR_BACKWARDS_COMPATIBILITY_MODE -g -Wall -Wextra -DOSATOMIC_USE_INLINED=1 -Ithird_party/abseil-cpp -Ithird_party/upb -Isrc/core/ext/upb-generated -O2 -Wframe-larger-than=16384 -fPIC -I. -Iinclude -I/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/gens -I/usr/local/include -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -arch i386 -arch x86_64  -Ithird_party/zlib -Ithird_party/cares -Ithird_party/cares/cares -fvisibility=hidden -D_GNU_SOURCE -Ithird_party/cares/config_darwin    -DWIN32_LEAN_AND_MEAN -D_HAS_EXCEPTIONS=0 -DNOMINMAX -DHAVE_CONFIG_H -std=c99  -g   -g -MMD -MF /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/cares/cares/ares__close_sockets.dep -c -o /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/cares/cares/ares__close_sockets.o third_party/cares/cares/ares__close_sockets.c
[C]       Compiling third_party/cares/cares/ares__get_hostent.c
mkdir -p `dirname /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/cares/cares/ares__get_hostent.o`
5 warnings generated.
gcc -Ithird_party/googletest/googletest/include -Ithird_party/googletest/googlemock/include -Ithird_party/boringssl-with-bazel/src/include -Ithird_party/cares -Ithird_party/cares/cares -Ithird_party/address_sorting/include -DGPR_BACKWARDS_COMPATIBILITY_MODE -g -Wall -Wextra -DOSATOMIC_USE_INLINED=1 -Ithird_party/abseil-cpp -Ithird_party/upb -Isrc/core/ext/upb-generated -O2 -Wframe-larger-than=16384 -fPIC -I. -Iinclude -I/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/gens -I/usr/local/include -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -arch i386 -arch x86_64  -Ithird_party/zlib -Ithird_party/cares -Ithird_party/cares/cares -fvisibility=hidden -D_GNU_SOURCE -Ithird_party/cares/config_darwin    -DWIN32_LEAN_AND_MEAN -D_HAS_EXCEPTIONS=0 -DNOMINMAX -DHAVE_CONFIG_H -std=c99  -g   -g -MMD -MF /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/cares/cares/ares__get_hostent.dep -c -o /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/cares/cares/ares__get_hostent.o third_party/cares/cares/ares__get_hostent.c
3 warnings generated.
2 warnings generated.
third_party/zlib/gzlib.c:252:9: error: implicit declaration of function 'lseek' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        LSEEK(state->fd, 0, SEEK_END);  /* so gzoffset() is correct */
        ^
third_party/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
third_party/zlib/gzlib.c:252:9: note: did you mean 'fseek'?
third_party/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdio.h:162:6: note: 'fseek' declared here
int      fseek(FILE *, long, int);
         ^
third_party/zlib/gzlib.c:258:24: error: implicit declaration of function 'lseek' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        state->start = LSEEK(state->fd, 0, SEEK_CUR);
                       ^
third_party/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
third_party/zlib/gzlib.c:359:9: error: implicit declaration of function 'lseek' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    if (LSEEK(state->fd, state->start, SEEK_SET) == -1)
        ^
third_party/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
third_party/zlib/gzlib.c:400:15: error: implicit declaration of function 'lseek' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        ret = LSEEK(state->fd, offset - state->x.have, SEEK_CUR);
              ^
third_party/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
third_party/zlib/gzlib.c:496:14: error: implicit declaration of function 'lseek' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    offset = LSEEK(state->fd, 0, SEEK_CUR);
             ^
third_party/zlib/gzlib.c:14:17: note: expanded from macro 'LSEEK'
#  define LSEEK lseek
                ^
[C]       Compiling third_party/cares/cares/ares__read_line.c
5 errors generated.
third_party/zlib/gzwrite.c:89:20: error: implicit declaration of function 'write' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
            writ = write(state->fd, strm->next_in, put);
                   ^
third_party/zlib/gzwrite.c:89:20: note: did you mean 'fwrite'?
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdio.h:165:9: note: 'fwrite' declared here
size_t   fwrite(const void * __restrict __ptr, size_t __size, size_t __nitems, FILE * __restrict __stream) __DARWIN_ALIAS(fwrite);
         ^
third_party/zlib/gzwrite.c:110:24: error: implicit declaration of function 'write' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                writ = write(state->fd, state->x.next, put);
                       ^
third_party/zlib/gzwrite.c:661:9: error: implicit declaration of function 'close' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    if (close(state->fd) == -1)
        ^
mkdir -p `dirname /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/cares/cares/ares__read_line.o`
make: *** [/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/gzlib.o] Error 1
make: *** Waiting for unfinished jobs....
3 errors generated.
third_party/zlib/gzread.c:35:15: error: implicit declaration of function 'read' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        ret = read(state->fd, buf + *have, get);
              ^
third_party/zlib/gzread.c:35:15: note: did you mean 'fread'?
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdio.h:158:9: note: 'fread' declared here
size_t   fread(void * __restrict __ptr, size_t __size, size_t __nitems, FILE * __restrict __stream);
         ^
make: *** [/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/gzwrite.o] Error 1
gcc -Ithird_party/googletest/googletest/include -Ithird_party/googletest/googlemock/include -Ithird_party/boringssl-with-bazel/src/include -Ithird_party/cares -Ithird_party/cares/cares -Ithird_party/address_sorting/include -DGPR_BACKWARDS_COMPATIBILITY_MODE -g -Wall -Wextra -DOSATOMIC_USE_INLINED=1 -Ithird_party/abseil-cpp -Ithird_party/upb -Isrc/core/ext/upb-generated -O2 -Wframe-larger-than=16384 -fPIC -I. -Iinclude -I/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/gens -I/usr/local/include -DNDEBUG -DINSTALL_PREFIX=\"/usr/local\" -arch i386 -arch x86_64  -Ithird_party/zlib -Ithird_party/cares -Ithird_party/cares/cares -fvisibility=hidden -D_GNU_SOURCE -Ithird_party/cares/config_darwin    -DWIN32_LEAN_AND_MEAN -D_HAS_EXCEPTIONS=0 -DNOMINMAX -DHAVE_CONFIG_H -std=c99  -g   -g -MMD -MF /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/cares/cares/ares__read_line.dep -c -o /Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/cares/cares/ares__read_line.o third_party/cares/cares/ares__read_line.c
third_party/zlib/gzread.c:651:11: error: implicit declaration of function 'close' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    ret = close(state->fd);
          ^
2 errors generated.
make: *** [/Users/brodock/.rvm/gems/ruby-2.7.2/gems/grpc-1.30.2/src/ruby/ext/grpc/objs/opt/third_party/zlib/gzread.o] Error 1
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/Users/brodock/.rvm/rubies/ruby-2.7.2/bin/$(RUBY_BASE_NAME)

extconf failed, exit code 1
```

Make sure you include information that can help us debug (full error message, exception listing, stack trace, logs).

See [TROUBLESHOOTING.md](https://github.com/grpc/grpc/blob/master/TROUBLESHOOTING.md) for how to diagnose problems better.

### Anything else we should know about your project / environment?

I've also tried with latest `1.32.0` and got similar failure

--- a/Makefile
+++ b/Makefile
@@ -2455,6 +2455,7 @@ PUBLIC_HEADERS_C += \
 LIBZ_OBJS = $(addprefix $(OBJDIR)/$(CONFIG)/, $(addsuffix .o, $(basename $(LIBZ_SRC))))
 
 $(LIBZ_OBJS): CFLAGS += -fvisibility=hidden
+$(LIBZ_OBJS): CPPFLAGS += -DHAVE_UNISTD_H
 
 $(LIBDIR)/$(CONFIG)/libz.a:  $(LIBZ_OBJS) 
 	$(E) "[AR]      Creating $@"

--- a/build_handwritten.yaml
+++ b/build_handwritten.yaml
@@ -216,6 +216,7 @@ defaults:
     LDFLAGS: -g
   zlib:
     CFLAGS: -fvisibility=hidden
+    CPPFLAGS: -DHAVE_UNISTD_H
 php_config_m4:
   deps:
   - grpc

